### PR TITLE
Fix actions without parameters

### LINKED ISSUE #32665: App crashes when connecting a model action to a dashboard button
### Describe the bug

100% reproducible on stats instance. Can not reproduce locally.
Possibly related to #32666 (reviews table seems to be the key).

image

### To Reproduce

1. Create a model based on reviews table from sample db
2. Enable actions in that model
3. Create a new dashboard
4. Add an action button to it
5. Try to connect update action of that model to your button

### Expected behavior

No crash

### Logs

_No response_

### Information about your Metabase installation

```JSON
Built on 2023-07-26
Branch: master
Hash: d687abe
```

### Severity

P1/P2

### PR DESCRIPTION
**Pull request by ranquild**
Fixes https://github.com/metabase/metabase/issues/32665

Based on backend clojure code implicit actions might not have `parameters` set https://github.com/metabase/metabase/blob/b7cac603ae785f901d76b13c3d839e153a01189b/src/metabase/models/action.clj#L282. Given that `parameters` are optional for `Dashboard` as well, I went ahead and made them optional in `Action` as well and fixed found issues. One of them is in `ActionDashcardSettings` which is exactly what is shown in the screenshot in the issue.

### CODE REVIEW

File: frontend/src/metabase/actions/components/ActionViz/ActionDashcardSettings.unit.spec.tsx [hunk 1]
```diff
@@ -456,6 +456,16 @@ describe("ActionViz > ActionDashcardSettings", () => {
     ).toBeInTheDocument();
   });
 
+  it("should be valid and not crash when the action does not have parameters", async () => {
+    const { closeSpy } = setup({
+      dashcard: createMockActionDashboardCard({
+        action: createMockQueryAction(),
+      }),
+    });
+    await userEvent.click(screen.getByRole("button", { name: "Done" }));
```
Comment (ranquild) [inline frontend/src/metabase/actions/components/ActionViz/ActionDashcardSettings.unit.spec.tsx:465]: The test checks that for the case where `parameters` are `undefined` there is no bug in `isFormInvalid` logic, i.e. the settings form is considered valid.

File: frontend/src/metabase/actions/hooks/use-action-form/use-action-form.ts [hunk 2]
```diff
@@ -22,27 +23,28 @@ type Opts = {
 };
 
 const INITIAL_VALUES = {};
+const DEFAULT_PARAMETERS: WritebackParameter[] = [];
```
Comment (ranquild) [inline frontend/src/metabase/actions/hooks/use-action-form/use-action-form.ts:26]: Extracted here to not break memoization

File: frontend/src/metabase/actions/components/ActionViz/ActionDashcardSettings.unit.spec.tsx [hunk 3]
```diff
@@ -456,6 +456,16 @@ describe("ActionViz > ActionDashcardSettings", () => {
     ).toBeInTheDocument();
   });
 
+  it("should be valid and not crash when the action does not have parameters", async () => {
+    const { closeSpy } = setup({
+      dashcard: createMockActionDashboardCard({
+        action: createMockQueryAction(),
```
Comment (kamilmielnik) [inline frontend/src/metabase/actions/components/ActionViz/ActionDashcardSettings.unit.spec.tsx:462]: :x: This test passes in `master`

```suggestion
        action: createMockQueryAction({ parameters: undefined }),
```

File: frontend/src/metabase/actions/components/ActionViz/ActionDashcardSettings.unit.spec.tsx [hunk 4]
```diff
@@ -456,6 +456,16 @@ describe("ActionViz > ActionDashcardSettings", () => {
     ).toBeInTheDocument();
   });
 
+  it("should be valid and not crash when the action does not have parameters", async () => {
```
Comment (kamilmielnik) [inline frontend/src/metabase/actions/components/ActionViz/ActionDashcardSettings.unit.spec.tsx:459]:
```suggestion
  it("should be valid and not crash when the action does not have parameters (metabase#32665)", async () => {
```

**Review by kamilmielnik — DISMISSED** [review]

**Review by kamilmielnik — APPROVED** [review]
:+1:

**Pull request merged**